test/e2e/react-compiler/react-compiler.test.ts

```diff
@@ -0,0 +1,27 @@
+import { nextTestSetup, FileRef } from 'e2e-utils'
+import { join } from 'path'
+
+describe.each(
+  ['default', process.env.TURBOPACK ? undefined : 'babelrc'].filter(Boolean)
+)('react-compiler %s', (variant) => {
+  const { next } = nextTestSetup({
+    files:
+      variant === 'babelrc'
+        ? __dirname
+        : {
+            app: new FileRef(join(__dirname, 'app')),
+            'next.config.js': new FileRef(join(__dirname, 'next.config.js')),
+          },
+
+    dependencies: {
+      'babel-plugin-react-compiler': '0.0.0-experimental-4690415-20240515',
```

Comment:
Do we want to bundle the compiler like we do with all the react packages? Or is this intentionally a thing the user installs manually and updates on their own?